### ISSUE TITLE
vue/v-on-handler-style add "neve" behaviour from old "vue/v-on-function-call"

### ISSUE THREAD

**Issue by DouglasCalora** [vuejs/eslint-plugin-vue#2749]
I updated my lib to v10 and saw some breaking changes...
I changed my old "vue/v-on-function-call"

```js
'vue/v-on-function-call': [
  'error',
  'never',
  {
    ignoreIncludesComment: false
  }
]
```

But I didn't find a similar behavior to "never", am I doing something wrong?

never behaviour:
```js
<template>
  <!-- ✓ GOOD -->
  <button v-on:click="closeModal">Close</button>
  <button v-on:click="closeModal(arg)">Close</button>

  <!-- ✗ BAD -->
  <button v-on:click="closeModal()">Close</button>
</template>
```

I have my component that now throw errors but before worked fine:
```html
<div class="flex justify-between q-mb-xl">
  <qas-btn
    label="Esqueci o e-mail"
    variant="tertiary"
    @click="redirectToAuthForgotten('email')"
  />

  <qas-btn
    label="Esqueci a senha"
    variant="tertiary"
    @click="redirectToAuthForgotten('password')"
  />
</div>
```
